src/fstests/README.md

```diff
@@ -0,0 +1,148 @@
+        ca-certificates debootstrap e2fslibs-dev ed fakechroot gettext git \
+        golang-1.13-go libblkid-dev libdbus-1-3 libgdbm-dev libicu-dev \
+        libkeyutils-dev libssl-dev libsystemd-dev libtool-bin liburcu-dev \
+        lsb-release meson pkg-config python3-setuptools rsync symlinks \
+        qemu-utils uuid-dev zlib1g-dev
+
+For building the Linux kernel:
+
+    sudo apt install -y bc bison cpio flex gcc kmod libssl-dev libelf-dev
+
+> **Note** As a follow-up improvement, the kernel builds could be done in
+> Docker containers like any regular build produced by KernelCI.  Building
+> natively is to simplify the process for the PoC.
+
+For more details, see the [quick start guide for
+`kvm-xfstests`](https://github.com/tytso/xfstests-bld/blob/master/Documentation/kvm-quickstart.md)
+
+
+## Setting up kvm-xfstests
+
+The next step is to have `kvm-xfstests` installed and available in the
+execution `$PATH`.
+
+Check out the code:
+
+    git clone https://github.com/tytso/xfstests-bld.git
+    cd xfstests-bld
+
+Download a pre-built KVM image:
+
+    wget -O test-appliance/root_fs.img \
+        https://www.kernel.org/pub/linux/kernel/people/tytso/kvm-xfstests/root_fs.img.i386
+
+Generate the `kvm-xfstests` launcher script:
+
+    make kvm-xfstests
+
+Install it in your bin directory and add it to the `$PATH` if necessary:
+
+    cp kvm-xfstests ~/bin
+    export PATH=$HOME/bin:$PATH
+
+Check it's in your path now:
+
+    $ kvm-xfstests --help
+    Usage: kvm-xfstests [<OPTIONS>] smoke|full
+    Usage: kvm-xfstests [<OPTIONS>] <test> ...
+    Usage: kvm-xfstests [<OPTIONS>] -g <group> ...
+    Usage: kvm-xfstests [<OPTIONS>] shell|maint
+    Usage: kvm-xfstests [<OPTIONS>] syz <repro>
+    [...]
+
+
+## Building a kernel for fstests
+
+A particular set of kernel config options need to be enabled for fstests.  This
+is automated by the `install-kconfig` command.  KernelCI automated jobs will be
+using a Linux kernel source tarball, but to do check things are set up
+correctly by hand we'll use a regular git checkout here.
+
+Get the Linux kernel source tree (lightweight checkout), outside of the
+`xfstests-bld` directory:
+
+    mkdir linux
+    cd linux
+    git init
+    git remote add origin \
+        https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
+    git fetch origin --depth=1 v5.19
+    git checkout FETCH_HEAD -b linux-5.19
+
+Generate the kernel config for fstests and build the kernel image (no modules):
+
+    kvm-xfstests install-kconfig
+    make -j$(nproc)
+
+
+## Running the tests
+
+Then it's all ready to run some tests.  For example, the `smoke` tests which
+should take under 30min to run:
+
+    kvm-xfstests smoke
+
+Or to run just a single set of tests, which should take about 1min:
+
+    kvm-xfstests generic/001
+
+To get the XML results file , run a shell in the VM and copy the file to the
+host:
+
+    $ kvm-xfstests shell
+    # cp /results/ext4/results-4k/results.xml /vtmp/
```

Comment:
is the `v` in vtmp a typo?

Reply:
No it's not a typo. Why? Did you not have `/vtmp`?

Reply:
no, I didn't.. but I just put it on another path and works fine.